transaction/std/src/transaction_builder.rs

```diff
@@ -3247,7 +3304,7 @@ pub mod transaction_builder_tests {
             let global_index = 42;
             let gift_code_tx_out_private_key = recover_onetime_private_key(
                 funding_output_public_key,
-                sender.spend_private_key(),
+                sender.view_private_key(),
```

Comment:
@jcape @isis-mc this is the first time I can remember that I had to fix a bug that was ultimately root-caused to mixing up spend and view private keys.

so yeah, mark one tally for "possible value add from type-safety around spend vs. view keys"